### PR TITLE
Add UseUniversalArgs

### PR DESCRIPTION
**Pull request by KKyang**

### CODE REVIEW

File: tensilelite/Tensile/SolutionStructs.py [hunk 1]
```diff
@@ -3527,12 +3527,9 @@ def subCheckLdsBlockSizePerPad(tc, idx):
       # TODO: support ONLL if necessary
       state["OptNoLoadLoop"] = 0
 
-    if not state["ProblemType"]["GroupedGemm"]:
-      if state["ProblemType"]["SupportUserArgs"]:
-        reject(state, "Currently only grouped gemm supports SupportUserArgs.")
-    if state["GlobalSplitU"] > 1:
-      if state["ProblemType"]["SupportUserArgs"] and state["_GlobalAccumulation"] != 'MultipleBufferSingleKernel':
-        reject(state, "Currently SupportUserArgs does not support GSU > 1.")
+    # if state["GlobalSplitU"] > 1:
```
Comment (hcman2) [inline tensilelite/Tensile/SolutionStructs.py:3530]: Why not remove this code?

Comment (KKyang) [reply tensilelite/Tensile/SolutionStructs.py:3530]: To remind me the restriction :)

**Comment by KKyang** [comment]
gfx94x CI passed

File: tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp [hunk 2]
```diff
@@ -744,9 +746,19 @@ namespace Tensile
         }
     }
 
-    template <bool T_Debug, typename KA>
-    void ContractionSolution::kernelArgs(KA& args, const ContractionProblemParameters& param) const
+    template <bool T_Debug, bool Legacy, typename KA>
+    void ContractionSolution::kernelArgs(uint32_t                            gemmCount,
+                                         uint32_t                            argType,
```
Comment (jichangjichang) [inline tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp:751]: What is argType? should we use enum for different type?

File: tensilelite/Tensile/SolutionStructs.py (hunk 1 above)
Comment (jichangjichang) [reply tensilelite/Tensile/SolutionStructs.py:3530]: if we don't have this restriction now, why did you want to keep it to remind you?

File: tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp (hunk 2 above)
Comment (KKyang) [reply tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp:751]: In stage 1 we only have 0 and 1. In stage 2 we'll have different types.

Comment (KKyang) [reply tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp:751]: Add comment in code

**Comment by KKyang** [comment]
gfx94x local test passed

**Review by aazz44ss — APPROVED** [review]

**Review by hcman2 — APPROVED** [review]

**Comment by Jinp800125** [comment]
Is UseUniversalArgs just for groupgemm? The default value could be false?

**Pull request merged**